src/core/Value.cc

```diff
@@ -269,6 +269,19 @@ bool Value::toBool() const
   // NOLINTEND(bugprone-branch-clone)
 }
 
+// Convert the value to a double with an integer value, for use in bitwise operations.
+// Since there are several possible ways to do this (floor, ceil, round) this function
+// centralizes the choice for consistency.
+double Value::toInteger() const
+{
+  return floor(this->toDouble());
```

Comment:
Now that we support negative numbers, I think this should trunc() here, not round down

Reply:
I'm not strongly opposed, but It seems like truncation is generally the wrong answer for float-to-integer conversion, because it means that zero is over-represented in the results.  If you want something that's more human-obvious for negative numbers than floor(), would you consider round()?

Reply:
It's not really conversion, it's just casting the result to an integer, and truncating is usually the answer.
For reference, Javascript and Python agrees. Python needs an explicit conversion, but it will trunc upon conversion.

Reply:
Yeah, I think they're wrong too.  But OK.  I can't come up with any practical cases where there's obviously-right or obviously-wrong answers, in the context of bitwise arithmetic.

(But see my just-filed doc bug for the ... odd... behavior of picking a random integer from a range that crosses zero, if you use trunc().  In that case, I'd say that trunc() is one of the most wrong answers.  However, I can't come up with any way to reasonably tie that to bitwise arithmetic.)

Reply:
Another alternative is to implement the bitwise operations for floating point values.  But few people would understand, and it would be an unpleasant amount of work.

Reply:
Array indexes appear to truncate, so that's one rationale.

Reply:
Least surprising when comparing to other implementations is also a good rationale. We probably violate that in a bunch of places, but I'd think mostly by accident, not because we did the research and designed it to fit a particular class of applications :)